src/sparseml/transformers/utils/helpers.py

```diff
@@ -224,3 +211,106 @@ def resolve_sequence_length(config: AutoConfig) -> int:
         "(inferred from HF transformers config) "
     )
     return sequence_length
+
+
+def resolve_recipe_application(
+    recipe: Union[str, Path, None], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    """
+    Resolve the recipe to apply to the model.
+    :param recipe: the recipe to apply to the model.
+        It can be one of the following:
+        - None
+        - a path to the recipe file
+        - name of the recipe file (e.g. "recipe.yaml")
+            (assumed to be stored in the model_path instead
+            of RECIPE_NAME)
+        - a string containing the recipe
+
+        If recipe is None, the function will try to
+        find the recipe in the model_path if model_path
+        is a local path. If model_path is a huggingface
+        model_id, the function will try to download the
+        recipe from huggingface.co.
+
+    :param model_path: the path to the model to load
+    :return: the resolved recipe
+    """
+
+    if recipe is None:
+        # if recipe is None -> still look for recipe.yaml in the model_path
+        recipe = os.path.join(model_path, RECIPE_NAME)
+        if os.path.isfile(recipe):
+            return recipe
+        else:
+            request = requests.get(os.path.join(HUGGINGFACE_ADDRESS, model_path))
+            if request.status_code == 200:
+                _LOGGER.info(
+                    "model_path is a huggingface model id. "
+                    f"Attempting to download recipe from {HUGGINGFACE_ADDRESS}"
+                )
+                recipe = hf_hub_download(repo_id=model_path, filename=RECIPE_NAME)
+                return recipe
+
+    elif os.path.isfile(recipe):
+        # recipe is a path to a recipe file
+        return _resolve_recipe_file(recipe, model_path)
+
+    elif os.path.isfile(os.path.join(model_path, recipe)):
+        # recipe is a name of a recipe file
+        recipe = os.path.join(model_path, recipe)
+        return _resolve_recipe_file(recipe, model_path)
+    elif isinstance(recipe, str):
+        # recipe is a string containing the recipe
+        _LOGGER.debug(
+            "Applying the recipe string directly to the model, without "
+            "checking for a potential existing recipe in the model_path."
+        )
+        return recipe
+
+    _LOGGER.info(
+        "No recipe requested and no default recipe "
+        f"found in {model_path}. Skipping recipe application."
+    )
+    return None
+
+
+def _resolve_recipe_file(
+    requested_recipe: Union[str, Path], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    default_recipe = os.path.join(model_path, RECIPE_NAME)
```

Comment:
If model path is a stub(containing a default recipe) , and requested recipe is a local file, won't this point to the wrong location?

Reply:
Yes, will account for that

Reply:
I still don't see this accounted for? As in we will display the wrong warning in that case

Reply:
@rahul-tuli took more time today to improve that logic. thank you for pointing out the flaws. please rereview